### PR TITLE
Finished Test API Using MySQL Database

### PR DESCRIPTION
**Pull request by Nico005**
Demo API.
Connects to a MySQL Database.
Creates a single table called users, you are able to do all the CRUD requests via an API

### Demo

Curl examples : 
#Get a list of all users in the database
curl http://<server_ip:port>/all_users

#Get a single user in database using the users id
curl http://<server_ip:port>/single_user/<users_id>

#Insert a User
curl -X POST -H "Content-Type: application/json" -d '{<columns and values>}' http://<server_ip:port>/create_new_user**
E.g : 
curl -X POST -H "Content-Type: application/json" -d '{"name": "Test User", "super_hero":"1"}' http://0.0.0.0:6543/create_new_user

#Update a User
curl -X PUT -H "Content-Type: application/json" -d '{<columns and values to be updated>}' http://<server_ip:port>/update_user/<user_id_to_update>

#Delete a User
curl -X DELETE http://<serverip:port>/delete_user/<user_id_to_delete>

### Notes

You will need to run the bootstrap script to install all requirements.
MySQL etc

## Testing Instructions

- Run the bootstrap script
- Run the setup file in the project    :  python setup.py develop
- Update the development.ini / production.ini connection details with your database credentials
- Run the initialize_battlecry_proj_db development.ini
- Start Project to test with the above curl examples pserve development.ini --reload

### CODE REVIEW

File: script/bootstrap [hunk 1]
```diff
@@ -78,4 +78,11 @@ fi
 echo "  Application Bootstrap 2/2. Ansible (allthethings)..."
 ansible-playbook ${playbook} --inventory-file=${inventory} --connection=local
 
-echo "Bootstrap Complete"
\ No newline at end of file
+echo " Install MySQL 5.7 "
+sudo apt-get install mysql-server-5.7
```
Comment (markdia) [inline script/bootstrap]: these things should be done in the provisioner - ansible

**Review by markdia — CHANGES_REQUESTED** [review]
I'm not sure how to actually review the git submodule.  There is probably an alternative solution to using a submodule that we should look at.

File: script/bootstrap (hunk 1 above)
Comment (oakensoul) [reply script/bootstrap]: Also, we should be using RDS to run a mysql server, you likely don’t want to install the server inside the vagrant box itself.

File: Vagrantfile [hunk 2]
```diff
@@ -5,8 +5,8 @@
 # vi: set ft=ruby :
 
 # UPDATE THESE FOR YOUR PROJECT
-project_name = "pyzappi"
-hosts_aliases = %w(pyzappi.localdomain "pyzappi.web-masons.org")
+project_name = "warcry"
+hosts_aliases = %w(warcry.localdomain "warcry.web-masons.org")
```
Comment (oakensoul) [inline Vagrantfile]: This is the Pyzapi project, so you don’t want to change the name of the project to something else. If you’re making a copy of the project, you could do that, but we don’t want to change the base project.

File: Vagrantfile [hunk 3]
```diff
@@ -46,12 +47,13 @@ Vagrant.configure(2) do |config|
   # the path on the host to the actual folder. The second argument is
   # the path on the guest to mount the folder. And the optional third
   # argument is a set of non-required options.
-  config.vm.synced_folder ".", "/vagrant",
+  config.vm.synced_folder ".", "/home/nico/",
```
Comment (oakensoul) [inline Vagrantfile]: We shouldn’t be changing the /vagrant folder. By changing it to a users folder that would mean all users in the future would have to share it.

File: Vagrantfile [hunk 4]
```diff
@@ -46,12 +47,13 @@ Vagrant.configure(2) do |config|
...
     :nfs => true,
     :mount_options => ['nolock,vers=3,udp,noatime,actimeo=1']
 
   config.vm.provider "virtualbox" do |vb|
-      vb.customize [ "guestproperty", "set", :id, "/VirtualBox/GuestAdd/VBoxService/--timesync-set-threshold", 1000 ]
+      vb.customize [ "guestproperty", "set", :id, "/VirtualBox/GuestAdd/VBoxService/--timesync-set-threshold", 2000 ]
```
Comment (oakensoul) [inline Vagrantfile]: Why are we changing the time threshold here? Was it causing issues? I’d love to understand a bit more.

File: script/bootstrap [hunk 5]
```diff
@@ -78,4 +78,11 @@ fi
...
+sudo apt-get install python-mysqldb
+
+echo " Install Virtuel Enviroment"
+sudo pip install virtualenv virtualenvwrapper
```
Comment (oakensoul) [inline script/bootstrap]: I have a good example for how to deal with the virtual env stuff. Let me see if I can load that up for you tonight.

**Review by oakensoul — CHANGES_REQUESTED** [review]
The one thing I’m concerned with is that in a few places, there are core changes to the base project that would bring it out of alignment of the goals of the project which is primarily that it is re-usable. If we’re going to place examples of “how to use it” in here, they should be inside an ‘/examples/‘ folder with some documentation for what the example shows.

Also, I noticed in your PR comments some of the ideas you had for your example, we should sit down and chat about some good REST API practices. I think I can offer you some direction that would be more in line with the REST API best practices. I can’t see your submodule from here, so it’s hard to see your implementation.

File: script/bootstrap (hunk 1 above)
Comment (Nico005) [reply script/bootstrap]: Ill take this out the script and connect MySQL on Amazon RDS

File: Vagrantfile (hunk 2 above)
Comment (Nico005) [reply Vagrantfile]: Will revert project name back to "pyzappi"

File: Vagrantfile (hunk 3 above)
Comment (Nico005) [reply Vagrantfile]: sorry this slipped through, was done as a test.. as was still getting the hang of vagrant :)
I will revert and sync to the vagrant folder

File: Vagrantfile (hunk 4 above)
Comment (Nico005) [reply Vagrantfile]: I had some network issues on my side, tested with the threshold of 1000 and seems fine now, will revert and commit

**Pull request closed without merging**